source/slang/slang-check-decl.cpp

```diff
@@ -9308,8 +9308,279 @@ namespace Slang
             this, funcDecl, attr, DeclAssociationKind::PrimalSubstituteFunc);
     }
 
+    struct LinearInheritanceGraphWithInfo
```

Comment:
There is already a getInheritanceInfo or similarly named function that return you a linear use list of type facets of a given type, including all transitive bases and extensions. We should just use that.

Reply:
I will switch the "getting inheritance" part of the logic over, the auxiliary processing + data-fetching will still need to be a separate pass.